lib/zendesk_api/collection.rb

```diff
@@ -332,6 +331,32 @@ def set_page_and_count(body)
       end
     end
 
+    def has_more_results?(response)
+      return false unless response["meta"].present? && response["results"].present?
+
+      response["meta"]["has_more"] && response["results"].length > 0
+    end
+
+    def get_response_body(link)
+      response_body = @client.connection.send("get", link).body
+
+      response_body
+    end
+
+    def get_next_page_data(original_response_body)
+      link = original_response_body["links"]["next"]
+
+      while link do
+        response = get_response_body(link)
+
+        original_response_body["results"] = original_response_body["results"] + response["results"]
```

Comment:
What types of data are we concatenating here?

Reply:
We are concatenating the data that each array has inside.